code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -186,6 +184,28 @@ Made by Xhuis
 						  under their command...</span>")
 	return 1
 
+/datum/game_mode/proc/remove_shadowling(datum/mind/ling_mind)
+	if(!istype(ling_mind) || !(ling_mind in shadows)) return 0
+	update_shadow_icons_removed(ling_mind)
+	shadows.Remove(ling_mind)
+	ling_mind.current.attack_log += "\[[time_stamp()]\] <span class='danger'>Deshadowlinged</span>"
+	ling_mind.special_role = null
+	for(var/obj/effect/proc_holder/spell/S in ling_mind.spell_list)
+		ling_mind.remove_spell(S)
+	var/mob/living/M = ling_mind.current
+	if(issilicon(M))
+		M.audible_message("<span class='notice'>[M] lets out a short blip.</span>", \
+						  "<span class='userdanger'>You have been turned into a robot! You are no longer a shadowling! Though you try, you cannot remember anything about your time as one...</span>")
+	else
+		M.visible_message("<span class='big'>[M] screams and contorts!</span>", \
+						  "<span class='userdanger'>THE LIGHT-- YOUR MIND-- <i>BURNS--</i></span>")
+		spawn(30)
+			M.visible_message("<span class='warning'>[M] suddenly bloats and explodes!</span>", \
```

Comment:
This isn't 20 questions for crying out loud, it's a code review.

@Xhuis you need to check that M is still a valid target before you send a visible message after than spawn. if(M && !qdeled(M)) should do the trick